src/frontend/screens/Sync/ProjectSyncDisplay.tsx

```diff
@@ -140,27 +141,38 @@ export const ProjectSyncDisplay = ({
     dataSyncEnabled: syncState.data.isSyncEnabled,
   });
 
-  // stops sync when user leaves sync screen.
-  // The api allows us to continue syncing even if the user is not on the sync screen, but for simplicity we are only allowing sync while on the sync screen.
-  // In the future we can easily enable background sync, there are just some UI questions that need to answered before we do that.
-  React.useEffect(() => {
-    const unsubscribe = navigation.addListener('beforeRemove', () => {
-      projectApi.$sync.stop();
-      queryClient.invalidateQueries({queryKey: [OBSERVATION_KEY, projectId]});
-    });
+  // Unset sync autostop when screen mounts
+  useFocusEffect(
+    React.useCallback(() => {
+      projectApi.$sync.setAutostopDataSyncTimeout(null);
+    }, [projectApi]),
+  );
 
-    return unsubscribe;
-  }, [navigation, projectApi, queryClient, projectId]);
+  const shouldAutostopSyncWhenLeavingScreen =
+    syncState.data.isSyncEnabled && syncStage.name === 'complete-full';
 
-  // TODO: Autostop data sync upon leaving screen while in specific state
-  // const shouldAutostopSyncWhenLeavingScreen = syncState.data.isSyncEnabled && syncStage.name === 'complete-full';
-  // React.useEffect(() => {
-  //   const unsubscribe = navigation.addListener('beforeRemove', () => {
-  //     projectApi.$sync.setAutostopDataSyncTimeout(30_000);
-  //   });
+  // Set up listener for autostopping sync (if applicable) and invalidating queries when navigating away from screen
+  useFocusEffect(
+    React.useCallback(() => {
+      const unsubscribe = navigation.addListener('beforeRemove', () => {
+        if (shouldAutostopSyncWhenLeavingScreen) {
+          projectApi.$sync.setAutostopDataSyncTimeout(30_000);
+        }
```

Comment:
Ah, okay. I _think_ the intended behavior is:

- If the screen is active, there is _no_ auto-stop timeout. The user must explicitly start and stop sync.
- If the screen is inactive, the auto-stop timeout should be 30 seconds. If the user had previously enabled sync, it will stop after 30 seconds of inactivity.

This seems like it doesn't do the second part if sync is enabled but incomplete. In that case, leaving the screen will never stop sync, even if it completes.

Reply:
for posterity since we talked about this elsewhere:

Evan's point about the expected behavior when sync finishes in the "background" (i.e. while you're not on the sync screen) is a valid one. Ideally in that case, we can set the autostop when that happens, but that logic would have to live closer to the SyncStore implementation rather than the sync screen specifically. also think that this issue could be alleviated by providing UI that exists in other parts of the app to indicate the progress of sync while you're on different screens, which I think is something that is being considered but not fully thought through yet.